common/evmcompat/helpers.go

```diff
@@ -102,6 +143,8 @@ func RecoverAddressFromTypedSig(hash []byte, sig []byte) (common.Address, error)
 			ssha.String("\x19TRON Signed Message:\n32"),
 			ssha.Bytes32(hash),
 		)
+	case SignatureType_BINANCE:
+		return BitcoinRecover(hash, sig[1:])
```

Comment:
`RecoverAddressFromTypedSig` is called from non-static Go contract methods, so this change may result in a consensus failure while a cluster is being upgraded, need to put this behind a feature flag. I would suggest changing the function signature to:
```
func RecoverAddressFromTypedSig(hash []byte, sig []byte, allowedSigTypes []SignatureType) (common.Address, error)
```
Then checking that `sig[0]` is in `allowedSigTypes`. And then doing a feature flag check in `Gateway.AddContractMapping` and `AddressMapper.AddIdentityMapping` to figure out what the `allowedSigTypes` should be.